app/src/main/java/co/nano/nanowallet/MainActivity.java

```diff
@@ -214,12 +209,12 @@ public void logOut(Logout logout) {
         // null out component
         mActivityComponent = null;
 
-        sharedPreferencesUtil.setConfirmedSeedBackedUp(false);
-        sharedPreferencesUtil.setFromNewWallet(false);
 
-        // go to the welcome fragment
-        getFragmentUtility().clearStack();
-        getFragmentUtility().replace(new IntroWelcomeFragment(), FragmentUtility.Animation.CROSSFADE);
+        //  if preferences have successfully been saved, go to the welcome fragment
+        if (sharedPreferencesUtil.setFromNewWallet(false) && sharedPreferencesUtil.setConfirmedSeedBackedUp(false)) {
```

Comment:
On the off chance that you get that you get an InterruptedException when writing to disk and false is returned from the sharedPreferencesUtil, wouldn't we see a blank screen here?

Reply:
I find that this hardly happens because both `apply()` and `commit()` write to the disk but, `commit()` just waits for a response. I could make the fallback resort to using the `apply()` method, and just launch the fragment either way?

Reply:
I agree that it doesn't happen very often but it does occasionally. I was wrong earlier about the blank screen since this is the logout method. However, since we are logging out, the fragment transaction is the visual cue that the user is logged out. I think it would beneficial do the fragment transaction regardless.

Reply:
Alright, with or without the call to the `setInBackground()` method?